src/libdrakvuf/win-processes.c

```diff
@@ -466,19 +466,36 @@ bool win_enumerate_processes_with_module(drakvuf_t drakvuf, const char* module_n
 
         vmi_pid_t pid;
         addr_t module_list_head;
+
         if ((win_get_process_pid(drakvuf, current_process, &pid) == VMI_SUCCESS) &&
                 win_get_module_list(drakvuf, current_process, &module_list_head))
         {
-            addr_t dllbase;
-            access_context_t ctx =
-            {
-                .translate_mechanism = VMI_TM_PROCESS_PID,
-                .pid = pid,
-            };
-            if (win_get_module_base_addr_ctx(drakvuf, module_list_head, &ctx, module_name, &dllbase))
+            access_context_t ctx = { .translate_mechanism = VMI_TM_PROCESS_DTB };
+
+            if ( vmi_pid_to_dtb( drakvuf->vmi, pid, &ctx.dtb ) == VMI_SUCCESS )
             {
-                if (visitor_func(drakvuf, current_process, visitor_ctx))
-                    return true;
+                module_info_t* module_info = win_get_module_info_ctx( drakvuf, module_list_head, &ctx, module_name );
+
+                if ( module_info )
+                {
+                    bool ret ;
+
+                    module_info->eprocess_addr = current_process ;
+                    module_info->dtb           = ctx.dtb ;
+                    module_info->pid           = pid ;
+
+                    ret = visitor_func( drakvuf, module_info, visitor_ctx );
+
+                    if ( module_info->full_name.contents )
+                        free( module_info->full_name.contents  );
```

Comment:
You could just use `g_free` instead of doing the if-then-free. `g_free` is NULL safe.

Reply:
Agreed

Reply:
Agreed